### PR TITLE
Flexible CTU Logging Interface

### PR DESCRIPTION
**Pull request by LoghinVladDev**

### CODE REVIEW

**Comment by LoghinVladDev** [comment]
Uncovered code is part of release. Most code smells are for:
- Compat with << with modifiers (std::endl)
- Release build vhaving no logging

File: src/core/lang/flag/FlagEnum.hpp [hunk 1]
```diff
@@ -0,0 +1,181 @@
+//
```
Comment (StefanPaulet) [inline src/core/lang/flag/FlagEnum.hpp:1]: Perhaps add documentation, so the function of FlagEnum becomes clearer.

File: test/unittests/LoggerTest.cpp [hunk 2]
```diff
@@ -0,0 +1,887 @@
...
+#ifndef NDEBUG
+TEST(LoggerTest, basicOut) {
+  stringstream outbuf;
+  auto logger = Logger::get(outbuf);
+  logger() << "basic string output, followed by numeric: " << 123 << std::hex << 15 << std::dec << 24;
+  ASSERT_TRUE(contains(outbuf, "basic string output, followed by numeric: 123f24"));
+}
+
+#define WRAP_ASSERT_TRUE(...)                                                                                          \
+  do {                                                                                                                 \
```
Comment (StefanPaulet) [inline test/unittests/LoggerTest.cpp:205]: Why the need for a do..while(false). Isn't a simple {} block the same?

File: src/core/logging/Logger.hpp [hunk 3]
```diff
@@ -0,0 +1,376 @@
...
+      _pOutput(&out), _filter(filterFlags & mask) {}
+
+  LoggerOutput(std::ostream& out, FilterFlagBits filterLevel) noexcept :
+      _pOutput(&out), _filter(static_cast<FilterFlags>(filterLevel)) {}
+
+  [[nodiscard]] constexpr auto& output() noexcept { return *_pOutput; }
+  [[nodiscard]] constexpr auto const& output() const noexcept { return *_pOutput; }
+
+  [[nodiscard]] constexpr auto allows(meta::LogLevelFlags levels) const noexcept {
+    return (_filter & levels & mask) != 0u;
```
Comment (StefanPaulet) [inline src/core/logging/Logger.hpp]: Is & mask necessary?

**Review by StefanPaulet — CHANGES_REQUESTED** [review]

File: test/unittests/LoggerTest.cpp (hunk 2 above)
Comment (LoghinVladDev) [reply test/unittests/LoggerTest.cpp:205]: Esthetical, forces use of `;` after function-like call

File: src/core/logging/Logger.hpp (hunk 3 above)
Comment (LoghinVladDev) [reply src/core/logging/Logger.hpp]: paranoia, and a double check doesn't hurt in this case
perhaps an assert is more fitting in this case

**Review by StefanPaulet — APPROVED** [review]

**Pull request merged**